lms/djangoapps/learner_dashboard/views.py

```diff
@@ -47,3 +57,65 @@ def program_details(request, program_uuid):
     }
 
     return render_to_response('learner_dashboard/program_details.html', context)
+
+
+class ProgramDiscussionIframeView(APIView, ProgramSpecificViewMixin):
+    """
+    A view for retrieving Program Discussion IFrame .
+
+    Path: ``/dashboard/programs/{program_uuid}/discussion/``
+
+    Accepts: [GET]
+
+    ------------------------------------------------------------------------------------
+    GET
+    ------------------------------------------------------------------------------------
+
+    **Returns**
+
+        * 200: OK - Contains a program discussion iframe.
+        * 401: The requesting user is not authenticated.
+        * 403: The requesting user lacks access to the program.
+        * 404: The requested program does not exist.
+
+    **Response**
+
+        In the case of a 200 response code, the response will be iframe HTML and status if discussion is configured
+        for the program.
+
+    **Example**
+
+        {
+            'enabled_for_masters': True,
+            'discussion': {
+                "iframe": "
+                            <iframe
+                                id='lti-tab-embed'
+                                style='width: 100%; min-height: 800px; border: none'
+                                srcdoc='{srcdoc}'
+                             >
+                            </iframe>
+                            ",
+                "enabled": false
+            }
+        }
+
+
+    """
+    authentication_classes = (BearerAuthentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated, IsEnrolledInProgram)
+
+    def get(self, request, program_uuid):
+        """ GET handler """
+        program_discussion_lti = ProgramDiscussionLTI(program_uuid, request)
+        return Response(
+
+            {
+                'enabled_for_masters': program_discussion_lti.is_enabled_for_masters,
+                'discussion': {
+                    'iframe': program_discussion_lti.render_iframe(),
+                    'enabled': program_discussion_lti.is_configured,
```

Comment:
I think this is not required for masters? we have "enabled_for_masters" in this case